__tests__/app.test.js

```diff
@@ -15,33 +17,195 @@ afterAll(() => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((result) => {
+        expect(result.body.comments.length).toBe(11);
+        expect(result.body.comments).toBeSortedBy("created_at", {
+          descending: true,
+          coerce: true,
+        });
+        result.body.comments.forEach((comment) => {
+          expect(typeof comment.votes).toBe("number");
+          expect(typeof comment.author).toBe("string");
+          expect(typeof comment.comment_id).toBe("number");
+          expect(typeof comment.body).toBe("string");
+          expect(typeof comment.article_id).toBe("number");
+          expect(typeof comment.created_at).toBe("string");
+        });
+      });
+  });
+
+  test("POST - status 201 - post a comment to article if article exists, responds with the posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .expect(201)
+      .send({
+        body: "I am adding a comment",
+        author: "icellusedkars",
+      })
+      .then((result) => {
+           const { comment } = result.body; 
+           expect(comment.comment_id).toBe(19);
+           expect(comment.body).toBe("I am adding a comment");
+           expect(comment.article_id).toBe(1);
+        expect(comment.author).toBe("icellusedkars");
+        expect(comment.votes).toBe(0)
+         expect(typeof comment.created_at).toBe('string');
+      });
+  });
+});
+
+describe("/api/articles/10000000000/comments", () => {
+  test('GET - status 404 - responds with error message "Article not found!"', () => {
+    return request(app)
+      .get("/api/articles/1000000000/comments")
+      .expect(404)
+      .then((result) => {
+        expect(result.body.msg).toBe("Article not found!");
+      });
+  });
+
+  test('POST - status 404 - responds with error message "Article not found!"', () => {
+    return request(app)
+      .post("/api/articles/1000000000/comments")
+      .send({
+        body: "I am adding a comment",
+        author: "icellusedkars",
+      })
+      .expect(404)
+      .then((result) => {
+        expect(result.body.msg).toBe("Article not found!");
+      });
+  });
+});
+
+describe("/api/articles/nonsense/comments", () => {
+  test('GET - status 400 - responds with error message "Bad request!"', () => {
+    return request(app)
+      .get("/api/articles/nonsense/comments")
+      .expect(400)
+      .then((result) => {
+        expect(result.body.msg).toBe("Bad request!");
+      });
+  });
+    test('POST - status 400 - responds with error message "Bad request!"', () => {
+      return request(app)
+        .post("/api/articles/nonsense/comments")
+        .send({
+          body: "I am adding a comment",
+          author: "icellusedkars",
+        })
+        .expect(400)
+        .then((result) => {
+          expect(result.body.msg).toBe("Bad request!");
+        });
+    });
```

Comment:
you need some error handling for a post request with missing properties on the body - this would be a 400 bad request.